py/server/deephaven/table.py

```diff
@@ -2469,32 +2494,31 @@ def keys(self) -> Table:
         else:
             return self.table.select_distinct(self.key_columns)
 
-    @property
+    @cached_property
     def unique_keys(self) -> bool:
         """Whether the keys in the underlying table must always be unique. If keys must be unique, one can expect
         that self.table.select_distinct(self.key_columns) and self.table.view(self.key_columns) operations always
         produce equivalent tables."""
-        if self._unique_keys is None:
-            self._unique_keys = self.j_partitioned_table.uniqueKeys()
-        return self._unique_keys
+        return self.j_partitioned_table.uniqueKeys()
 
-    @property
+    @cached_property
     def constituent_column(self) -> str:
         """The name of the column containing constituent tables."""
-        if self._constituent_column is None:
-            self._constituent_column = self.j_partitioned_table.constituentColumnName()
-        return self._constituent_column
+        return self.j_partitioned_table.constituentColumnName()
+
+    @cached_property
+    def constituent_table_definition(self) -> TableDefinition:
+        """The column definitions for constituent tables. All constituent tables in a partitioned table have the
+        same column definitions."""
+        return TableDefinition(self.j_partitioned_table.constituentDefinition())
 
     @property
```

Comment:
Also cacheable.

Reply:
Ditto. In this case, it's a purely python operation over a dict. The dict itself is cached.